clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11857,6 +11857,9 @@ def err_sycl_add_ir_attribute_invalid_value : Error<
 def err_sycl_add_ir_attribute_invalid_filter : Error<
   "initializer list in the first argument of %0 must contain only string "
   "literals">;
+def warn_sycl_old_and_new_kernel_attributes : Warning<
+  "kernel has attribute %0 but also has kernel properties. Conflicting "
+  "properties are ignored">, InGroup<IgnoredAttributes>;
```

Comment:
I discussed this with @Pennycook and @rolandschulz. The conclusion we came to was that in general the users should use one or the other, so having a warning if there is a potential for conflict should be sufficient. The other solution would require the FE to recognize names inside `add_ir_attributes_function`, parse their values, then compare. In general we have been trying to avoid the frontend needing to do property-specific parsing.

Reply:
> Are all properties specified by add_ir_attributes_function ignored or only conflicting ones? What does ignore mean actually?

It only ignores the conflicting ones. "Ignoring" in this case refers to the properties not having the requested effect. For example, for `sub_group_size`, the `reqd_sub_group_size` will dictate the sub-group size of the kernel and the `sub_group_size` property will have no effect.

> From a FE perspective it looks like that properties are attached to kernel irrespective of any possible conflict. So I assume by ignore you mean something further along compilation process?

Indeed. The FE should be no wiser to the effect of the ignored properties. `sycl-post-link` will see that FE has already generated the corresponding metadata and will just skip the generated LLVM attribute.